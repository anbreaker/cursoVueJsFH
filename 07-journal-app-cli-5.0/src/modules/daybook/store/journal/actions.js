// export const myActions = async ({ commit })=>{

// }

export const loadEntries = async ({ commit }) => {
  //
};

export const createEntries = async ({ commit }) => {
  //
};

export const updateEntries = async ({ commit }) => {
  //
};
