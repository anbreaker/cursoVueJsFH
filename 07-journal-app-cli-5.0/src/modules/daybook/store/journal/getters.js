// export const myGetters = (state)=>{
// 	return state.myState
// }
