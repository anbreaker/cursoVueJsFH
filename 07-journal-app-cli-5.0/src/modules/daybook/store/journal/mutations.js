// export const myMutations = (state)=>{

// }

export const setEntries = (state, entries) => {
  state.entries = [...state.entries, ...entries];

  state.isLoading = false;
};

export const addEntry = (state, entry) => {
  state.entries = [entry, ...state.entries];
};

export const updateEntry = (state, entry) => {
  const idx = state.entries.map((entry) => entry.id).indexOf(entry.id);

  state.entries[idx] = entry;
};
