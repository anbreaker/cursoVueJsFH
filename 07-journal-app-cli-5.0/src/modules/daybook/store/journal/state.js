export default () => ({
  islocal: true,
  entries: [
    {
      id: new Date().getTime(), // unique id for each entry
      date: new Date().toDateString(), //sat 23 jun
      text: 'Rodrigor tiene musho peligro diodeno apetecan. Llevame al sircoo hasta luego Lucas jarl qué dise usteer caballo blanco caballo negroorl.',
      picture: null, // 'https://picsum.photos/200/300',
    },
    {
      id: new Date().getTime() + 100, // unique id for each entry
      date: new Date().toDateString(), //sat 23 jun
      text: 'Diodeno llevame al sircoo apetecan a peich papaar papaar no te digo trigo por no llamarte Rodrigor por la gloria de mi madre.',
      picture: null, // 'https://picsum.photos/200/300',
    },
    {
      id: new Date().getTime() + 200, // unique id for each entry
      date: new Date().toDateString(), //sat 23 jun
      text: 'Ahorarr se calle ustée al ataquerl a wan diodeno de la pradera qué dise usteer ahorarr torpedo amatomaa diodeno.',
      picture: null, // 'https://picsum.photos/200/300',
    },
  ],
});
