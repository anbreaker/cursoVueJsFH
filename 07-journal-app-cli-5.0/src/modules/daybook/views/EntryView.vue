<template>
  <template v-if="entry">
    <div class="entry-title d-flex justify-content-left align-items-baseline p-2">
      <span class="text-success fs-3 fw-bold">{{ dayMonthYear.day }}</span>
      <span class="mx-1 fs-3">{{ dayMonthYear.month }}</span>
      <span class="mx-2 fs-4 fw-light">{{ dayMonthYear.yearDay }}</span>
    </div>
    <div class="">
      <button class="btn btn-danger mx-2">
        <i class="fa fa-trash-alt"></i>
        Delete
      </button>
      <button class="btn btn-secondary text-white mx-2">
        <i class="fa fa-upload"></i>
        Upload Image
      </button>
    </div>

    <hr />
    <div class="d-flex flex-column px-3 h-75">
      <textarea
        class="area-color"
        placeholder="What happened?"
        id=""
        cols="30"
        rows="10"
        v-model="entry.text"
      ></textarea>
    </div>

    <Fab icon="fa-save" />

    <img
      class="img-thumbnail img-position"
      src="https://img.myloview.es/fotomurales/encinas-en-el-cerro-mingamorena-pelahustan-toledo-espana-europa-700-212590059.jpg"
      alt="entry picture"
    />
  </template>
</template>

<script>
  import { defineAsyncComponent } from 'vue';
  import { mapGetters } from 'vuex';

  import { getDayMonthYear } from '../helpers/getDayMonthYear';

  export default {
    props: {
      id: {
        type: String,
        required: true,
      },
    },

    components: {
      Fab: defineAsyncComponent(() => import('../components/Fab')),
    },

    data() {
      return {
        entry: null,
      };
    },

    computed: {
      ...mapGetters('journal', ['getEntryById']),

      dayMonthYear() {
        const { day, month, yearDay } = getDayMonthYear(this.entry.date);

        return {
          day,
          month,
          yearDay,
        };
      },
    },

    methods: {
      loadEntry() {
        const entry = this.getEntryById(this.id);

        if (!entry) return this.$router.push({ name: 'no-entry' });

        this.entry = entry;
      },
    },

    created() {
      this.loadEntry();
    },

    watch: {
      id() {
        this.loadEntry();
      },
    },
  };
</script>

<style lang="scss" scoped>
  .area-color {
    background: #343a40;
    border: none;
    color: aliceblue;
    font-size: 1.2rem;
    padding: 0.8rem;
    resize: none;

    &:focus {
      outline: none;
    }
  }

  .img-position {
    bottom: 125px;
    box-shadow: 0px 5px 10px rgba(0, 0, 0, 0.5);
    position: fixed;
    right: 20px;
    width: 200px;
  }
</style>
