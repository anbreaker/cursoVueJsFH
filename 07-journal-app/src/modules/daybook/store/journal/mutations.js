// export const myMutation = (state) => {

// }

export const setEntries = (state, entries) => {
  state.entries = [...state.entries, ...entries]

  state.isLoading = false
}

export const updateEntry = (state, entry) => {
  const idx = state.entries.map((itm) => itm.id).indexOf(entry.id)

  state.entries[idx] = entry
}

export const addEntry = (state, entry) => {
  // state -> entries -> la nueva entrada al inicio del array
  state.entries = [entry, ...state.entries]
}

export const deleteEntry = (state, id) => {
  state.entries = state.entries.filter((entry) => entry.id !== id)
}
