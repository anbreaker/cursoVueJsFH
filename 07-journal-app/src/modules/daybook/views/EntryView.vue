<template>
  <template v-if="entry">
    <div class="entry-title d-flex justify-content-between p-2">
      <div>
        <span class="text-success fs-3 fw-bold">{{ dayMonthYear.day }}</span>
        <span class="mx-1 fs-3">{{ dayMonthYear.month }}</span>
        <span class="mx-2 fs-4 fw-light">{{ dayMonthYear.yearDay }}</span>
      </div>

      <div>
        <button class="btn btn-danger mx-2">
          Delete
          <i class="fa fa-trash-alt"></i>
        </button>
        <button class="btn btn-primary">
          Update Image
          <i class="fa fa-upload"></i>
        </button>
      </div>
    </div>

    <hr />

    <div class="d-flex flex-column px-3 h-75">
      <textarea placeholder="what happened today?" v-model="entry.text">
      </textarea>
    </div>
  </template>

  <FloatingActionButton icon="fa-save" @on:click="saveEntry" />

  <img
    src="https://tipsmake.com/data1/thumbs/how-to-extract-img-files-in-windows-10-thumb-bzxI4IDgg.jpg"
    alt="Entry Picture"
    class="img-thumbnail"
  />
</template>

<script>
import { defineAsyncComponent } from "@vue/runtime-core";
import { mapActions, mapGetters } from "vuex";

import getMonthYear from "../helpers/getDayMonthYear";

export default {
  name: "EntryView",
  props: {
    id: {
      type: String,
      required: true
    }
  },

  components: {
    FloatingActionButton: defineAsyncComponent(() =>
      import("../components/FloatingActionButton.vue")
    )
  },

  data() {
    return {
      entry: null
    };
  },

  methods: {
    ...mapActions("journal", ["updateEntry"]),

    async saveEntry() {
      console.log("Save entry...");

      // dispacht action Journal Module
      // console.log(this.entry);
      this.updateEntry(this.entry);
    },

    loadEntry() {
      const entry = this.getEntryById(this.id);

      if (!entry) return this.$router.push({ name: "no-entry" });

      this.entry = entry;
    }
  },

  computed: {
    ...mapGetters("journal", ["getEntryById"]),

    dayMonthYear() {
      const { day, month, yearDay } = getMonthYear(this.entry.date);

      return { day, month, yearDay };
    }
  },

  watch: {
    id() {
      this.loadEntry();
    }
  },

  created() {
    // console.log(this.$route.params.id)
    this.loadEntry();
  }
};
</script>

<style lang="scss" scoped>
textarea {
  font-size: 20px;
  border: none;
  height: 100%;

  &:focus {
    outline: none;
  }
}

img {
  width: 200px;
  position: fixed;
  bottom: 150px;
  right: 20px;
  box-shadow: 0px 5px 10px rgba($color: #000000, $alpha: 0.2);
}
</style>