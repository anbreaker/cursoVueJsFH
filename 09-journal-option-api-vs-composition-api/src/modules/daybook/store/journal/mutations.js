// export const myMutations = (state)=>{

// }

export const setEntries = (state, entries) => {
  state.entries = [...state.entries, ...entries];

  state.isLoading = false;
};

export const addEntry = (state, entry) => {
  state.entries = [entry, ...state.entries];
};

export const updateEntry = (state, entry) => {
  const idx = state.entries.map((entry) => entry.id).indexOf(entry.id);

  state.entries[idx] = entry;
};

export const deleteEntry = (state, id) => {
  state.entries = state.entries.filter((entry) => entry.id !== id);
};

export const clearEntries = (state) => {
  state.entries = [];
};
