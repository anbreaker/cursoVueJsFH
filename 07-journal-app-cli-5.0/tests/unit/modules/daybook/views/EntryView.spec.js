import { createStore } from 'vuex';
import { shallowMount } from '@vue/test-utils';

import EntryView from '@/modules/daybook/views/EntryView.vue';

import journal from '@/modules/daybook/store/journal';
import { journalState } from '../../../../mock-data/test-journal-state';

const createVuexStore = (initialState) =>
  createStore({
    modules: {
      journal: {
        ...journal,
        state: { ...initialState },
      },
    },
  });

describe('testing EntryView Component', () => {
  const store = createVuexStore(journalState);

  const mockRouter = {
    push: jest.fn(),
  };

  let wrapper;

  beforeEach(() => {
    jest.clearAllMocks();

    wrapper = shallowMount(EntryView, {
      props: {
        id: journalState.entries[1].id,
      },

      global: {
        mocks: {
          $router: mockRouter,
        },
        plugins: [store],
      },
    });
  });

  test('must remove the user because the id does not exist', () => {
    const wrapper = shallowMount(EntryView, {
      props: {
        id: 'This id not exists',
      },

      global: {
        mocks: {
          $router: mockRouter,
        },
        plugins: [store],
      },
    });

    expect(mockRouter.push).toHaveBeenCalledWith({ name: 'no-entry' });
  });

  test('must display the entry correctly', () => {
    expect(wrapper.html()).toMatchSnapshot();

    expect(mockRouter.push).not.toHaveBeenCalled();
  });
});
