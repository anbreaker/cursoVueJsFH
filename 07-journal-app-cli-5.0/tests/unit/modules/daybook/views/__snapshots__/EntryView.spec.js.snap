// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing EntryView Component must display the entry correctly 1`] = `
<div class="entry-title d-flex justify-content-left align-items-baseline p-2"><span class="text-success fs-3 fw-bold">23</span><span class="mx-1 fs-3">July</span><span class="mx-2 fs-4 fw-light">2021, Friday</span></div>
<div class=""><input type="file" accept="image/png, image/jpeg, image/jpg" style="display: none;"><button class="btn btn-danger mx-2"><i class="fa fa-trash-alt"></i> Delete </button><button class="btn btn-secondary text-white mx-2"><i class="fa fa-upload"></i> Upload Image </button></div>
<hr>
<div class="d-flex flex-column px-3 h-75"><textarea class="area-color" placeholder="What happened?" id="" cols="30" rows="10"></textarea></div>
<fab-stub icon="fa-save"></fab-stub>
<img src="https://res.cloudinary.com/dlayuetwi/image/upload/v1655396687/curso-vue-fh/axrcrwtxo6uzzclm8upx.png" class="img-thumbnail img-position" alt="entry picture">
<!--v-if-->
`;
